url/src/lib.rs

```diff
@@ -2720,7 +2720,7 @@ impl Url {
                 _ => return Err(()),
             };
 
-            return file_url_segments_to_pathbuf(host, segments);
+            return file_url_segments_to_pathbuf(self.as_str().len(), host, segments);
```

Comment:
Please split the `estimated_len` into its own temp variable so it's cleaner to read what this is.

I think this over-alloc 7 bytes in 99% of cases on unix (`file://`). On redox, just two (`//`). On Windows, even 8 (`file:///`). Windows also has `hostname` support though, so to ensure no-realloc in most cases, you should probably do -6 there too (`file:/` - the extra `/` is for the leading `\` in hostname cases).

The cost of doing this calculation may be too much perf cost though - so please check if this minor memory improvement wouldn't regress perf too much.

Reply:
It adds a few nanoseconds. Worth it to use less memory IMO.

Reply:
Agreed.